### ISSUE TITLE
Warning when building with iPhone OS 4.0 beta 3

### ISSUE THREAD

**Issue by GoogleCodeExporter** [huangwei1983/accelerometer-simulator#2]
```
What steps will reproduce the problem?
1. Build AccelerometerSimulator

What is the expected output? What do you see instead?
1 Warning on the line of code that reads:
[notificationPort setDelegate:self];
"Class 'AccelerometerSimulation' does not implement the 'NSMachPortDelegate' 
protocol"

What version of the product are you using? On what operating system?

```

Original issue reported on code.google.com by `sonic.bo...@gmail.com` on 30 May 2010 at 8:31

**Comment by GoogleCodeExporter** [comment]
```
Anyone solved this???
Same problem. 10.6.6 xcode: Version 3.2.5

Really want this to work.. 
```

Original comment by `IanD...@gmail.com` on 12 Mar 2011 at 4:37

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Works now but still get compile error..  Funny I didn't change a thing. 
```

Original comment by `IanD...@gmail.com` on 14 Mar 2011 at 5:37

**Comment by GoogleCodeExporter** [comment]
```
Are you doing any special order to get the data sent to the simulator?
```

Original comment by `joe.thom...@gmail.com` on 27 Apr 2011 at 7:54

**Comment by GoogleCodeExporter** [comment]
```
This happened to me when I failed to add the QuartzCore.framework to my 
project.  What confused me was that the QuartzCore.framework was being added 
for something completely unrelated to the AccelerometerSimulator (which had 
previously been working in my project).

Hope this helps you.

```

Original comment by `cogitoen...@gmail.com` on 5 Jun 2011 at 5:30

**Comment by GoogleCodeExporter** [comment]
```
1 Warning on the line of code that reads:
[notificationPort setDelegate:self];
"Class 'AccelerometerSimulation' does not implement the 'NSMachPortDelegate' 
protocol"

the same problem to me
```

Original comment by `toh.pen...@gmail.com` on 18 Oct 2011 at 1:12

**Comment by GoogleCodeExporter** [comment]
```
how to figure it out ,please help me,thank you
```

Original comment by `toh.pen...@gmail.com` on 18 Oct 2011 at 1:12

**Comment by GoogleCodeExporter** [comment]
```
Changed code to iOS 5.0 compatibility, and fixed iOS 4.0+ warnings
```

Original comment by `mauri...@vetsmart.com.br` on 29 Jan 2012 at 10:33
- Changed state: **Fixed**
